packages/web-components/fast-foundation/src/dialog/dialog.ts

```diff
@@ -208,14 +221,52 @@ export class Dialog extends FoundationElement {
 
         if (bounds.length > 0) {
             bounds[0].focus();
+        } else {
+            if (this.dialog instanceof HTMLElement) {
+                this.dialog.focus;
+            }
         }
     };
 
     /**
      * we should only focus if focus has not already been brought to the dialog
      */
     private shouldForceFocus = (currentFocusElement: Element | null): boolean => {
-        return !this.hidden && !this.contains(currentFocusElement);
+        return this.isTrappingFocus && !this.contains(currentFocusElement);
+    };
+
+    /**
+     * we should we be active trapping focus
+     */
+    private shouldTrapFocus = (): boolean => {
+        return this.trapFocus && !this.hidden;
+    };
+
+    /**
+     *
+     *
+     * @internal
+     */
+    private updateTrapFocus = (shouldTrapFocusOverride?: boolean): void => {
+        const shouldTrapFocus =
+            shouldTrapFocusOverride === undefined
+                ? this.shouldTrapFocus()
+                : shouldTrapFocusOverride;
+
+        if (shouldTrapFocus && !this.isTrappingFocus) {
+            this.isTrappingFocus = true;
+            // Add an event listener for focusin events if we are trapping focus
+            document.addEventListener("focusin", this.handleDocumentFocus);
```

Comment:
This looks like it could be calling updateTrapFocus() more than once in this class. On connectedCallback, on change and on trap change. It's probably optimize since it should have the same handler signature but it might be safer if you call document.removeEventListener at the top of this method to ensure that it isn't being added multiple times. And then you won't need to call it on line 268 also.

Reply:
The ```this.isTrappingFocus``` flag ensures that we only ever have one listener as we check before adding/removing it.  The idea is to ensure that a dialog component that exists in the DOM but is hidden isn't running unnecessary code on every document focus event.